File: src/models/task.py
from enum import StrEnum

from pydantic import BaseModel


class Status(StrEnum):
    IN_PROGRESS = "in_progress"
    COMPLETED = "completed"
    FAILED = "failed"


class TaskModel(BaseModel):
    status: Status


class FullTaskModel(BaseModel):
    task_id: int
    status: Status

File: frontend/src/models/task.ts
export const Status = {
  IN_PROGRESS: "in_progress",
  COMPLETED: "completed",
  FAILED: "failed",
} as const;

export type Status = (typeof Status)[keyof typeof Status];

export interface TaskModel {
  status: Status;
  theme: string;
}

export interface FullTaskModel {
  task_id: number;
  status: Status;
  theme: string;
}
